include/swift/Runtime/Config.h

```diff
@@ -216,6 +216,7 @@
 // But some of the runtime entries are invoked directly from
 // the foundation. Therefore they should be visible.
 #define SWIFT_RT_ENTRY_VISIBILITY SWIFT_RUNTIME_EXPORT
+#define SWIFT_EXTERN_RT_ENTRY_VISIBILITY SWIFT_EXTERN_RUNTIME_EXPORT
```

Comment:
I think a better way to make intentions explicit there would be to have a `SWIFT_CALLED_VIA_SILGEN_NAME` marker. That would incorporate `extern "C"` and potentially be useful for keeping the C++ side and the Swift side in sync.